tools/applets/dds_channel.py

```diff
@@ -0,0 +1,249 @@
+                            QtWidgets.QSizePolicy.Fixed)
+        label.setAlignment(QtCore.Qt.AlignHCenter)
+        label.setFont(titlefont)
+        grid.addWidget(label, 0, 0, 1, 3)
+
+        label = QtWidgets.QLabel("Frequency (MHz)")
+        label.setSizePolicy(QtWidgets.QSizePolicy.Fixed,
+                            QtWidgets.QSizePolicy.Fixed)
+        label.setFont(labelfont)
+        grid.addWidget(label, 1, 0)
+
+        label = QtWidgets.QLabel("Amplitude")
+        label.setSizePolicy(QtWidgets.QSizePolicy.Fixed,
+                            QtWidgets.QSizePolicy.Fixed)
+        label.setFont(labelfont)
+        grid.addWidget(label, 1, 1)
+
+        self.freq_box = QtWidgets.QDoubleSpinBox()
+        self.freq_box.setDecimals(3)
+        self.freq_box.setMinimum(1.0)
+        self.freq_box.setMaximum(500.0)
+        self.freq_box.setSingleStep(0.1)
+        self.freq_box.setFont(spinboxfont)
+        self.freq_box.setKeyboardTracking(False)
+        MHz_to_Hz = 1.e6
+        self.freq_box.setValue(self.dds_parameters.frequency / MHz_to_Hz)
+        grid.addWidget(self.freq_box, 2, 0)
+
+        self.amp_box = QtWidgets.QDoubleSpinBox()
+        self.amp_box.setDecimals(5)
+        self.amp_box.setMinimum(0.0)
+        self.amp_box.setMaximum(1.0)
+        self.amp_box.setSingleStep(0.01)
+        self.amp_box.setFont(spinboxfont)
+        self.amp_box.setSizePolicy(QtWidgets.QSizePolicy.Preferred,
+                                   QtWidgets.QSizePolicy.Preferred)
+        self.amp_box.setKeyboardTracking(False)
+        self.amp_box.setValue(self.dds_parameters.amplitude)
+        grid.addWidget(self.amp_box, 2, 1)
+
+        self.switch_button = QtWidgets.QPushButton("o")
+        self.switch_button.setFont(buttonfont)
+        self.switch_button.setSizePolicy(QtWidgets.QSizePolicy.Fixed,
+                                         QtWidgets.QSizePolicy.Fixed)
+        self.switch_button.setCheckable(True)
+        if self.dds_parameters.state:
+            self.switch_button.setChecked(self.dds_parameters.state)
+            self.set_switch_button_text(self.dds_parameters.state)
+        grid.addWidget(self.switch_button, 2, 2)
+
+    def setup_gui_listeners(self):
+        self.freq_box.valueChanged.connect(self.on_widget_freq_changed)
+        self.amp_box.valueChanged.connect(self.on_widget_amp_changed)
+        self.switch_button.clicked.connect(self.on_widget_switch_changed)
+
+    def on_widget_freq_changed(self, val):
+        MHz_to_Hz = 1.e6
+        self.dds_parameters.set_frequency(val * MHz_to_Hz)
+
+    def on_widget_amp_changed(self, val):
+        self.dds_parameters.set_amplitude(val)
+
+    def on_widget_switch_changed(self, checked):
+        self.dds_parameters.set_state(checked)
+        self.set_switch_button_text(checked)
+
+    def on_monitor_freq_changed(self, val):
+        MHz_to_Hz = 1.e6
+        self.freq_box.blockSignals(True)
+        self.freq_box.setValue(val / MHz_to_Hz)
```

Comment:
Perhaps a unit conversion module would be good.

Reply:
Agreed. There is some discussion in https://github.com/Jayich-Lab/pydux/issues/1629 about how to do unit conversions.